### PR TITLE
refactored for issue#1

### LINKED ISSUE #1: Review
**Contracts**

- The `claim` function can be made `external` to save gas. https://github.com/ArchanaVaideeswaran/crypto-airdrop/blob/0dec0a0bb409eda94e519b02485c7727c0e2e4be/contracts/MerkleAirdrop.sol#L19
- Use the compiler optimization options such as enabling optimizations with the default 200 runs. https://github.com/ArchanaVaideeswaran/crypto-airdrop/blob/0dec0a0bb409eda94e519b02485c7727c0e2e4be/hardhat.config.ts#L10 References: [hardhat.config.ts examples](https://www.programcreek.com/typescript/?api=hardhat/types.HardhatUserConfig)

**Code formatting**
- Use prettier code formatter (vs-code extension) to format the .sol and .js files of the project.

**Deployment**
- It is better to use the `truffle dashboard` plugin to deploy the contracts instead of pasting or using private keys. Try not to use private keys anywhere in the project.

**Packages and config**
- If `@nomicfoundation/hardhat-toolbox` is installed then it is not necessary to install and import`@nomiclabs/hardhat-etherscan` in the `hardhat-config.ts`. [hardhat-toolbox](https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-toolbox)

### PR DESCRIPTION
**Pull request by archana-vy**

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
-  networks: {
-    localhost: {
-      url: "http://127.0.0.1:8545/",
+    solidity: {
+        compilers: [
+            {
+                version: "0.8.17",
+                settings: {
+                    optimizer: { enabled: true, runs: 200 },
+                    evmVersion: "istanbul",
```
Comment (sumit03guha) [inline hardhat.config.ts]: no need to mention the evmVersion explicitly. The default and the latest one is `london`

**Review by sumit03guha — CHANGES_REQUESTED** [review]
remove the `evmVersion` key from hardhat config file.

File: hardhat.config.ts [hunk 2]
```diff
@@ -1,36 +1,34 @@
...
-  networks: {
-    localhost: {
-      url: "http://127.0.0.1:8545/",
+    solidity: {
+        compilers: [
+            {
+                version: "0.8.17",
+                settings: {
+                    optimizer: { enabled: true, runs: 200 },
+                },
```
Comment (archana-vy) [inline hardhat.config.ts:15]: removed evmVersion

**Comment by sumit03guha** [comment]
LGTM

**Pull request merged**